File: BOJ/1094/1094막대기.py
stick = 64
count = 0
sum_stick = 0

X = int(input())

#루프를 X와 같을 때까지 실행
while sum_stick != X:

    #현재 막대와 합이 X보다 크면 막대를 절반으로 자름
    if((stick + sum_stick) > X):
        stick = stick // 2
        continue

    sum_stick += stick
    count += 1
    stick = stick // 2
    
print(count)

File: 백준/ETC/1094/1094막대기.cpp
#include <iostream>
using namespace std;

int main() {
	int stick = 64, X, count = 0;
	cin >> X;

	int sum_stick = 0;

	
	while (sum_stick != X){

		if (stick + sum_stick > X) {
			stick /= 2;
			continue;
		}

		sum_stick += stick;
		count++;
		stick /= 2;

	}
	cout << count;
	
	return 0;
}
